fiat-core/src/main/java/com/netflix/spinnaker/fiat/model/UserPermission.java

```diff
@@ -131,10 +136,16 @@ public View(UserPermission permission) {
         val set =
             this.extensionResources.computeIfAbsent(
                 resource.getResourceType(), (ignore) -> new HashSet<>());
-        val view = ((Viewable) resource).getView(permission.getRoles(), permission.isAdmin());
-        if (view instanceof Authorizable) {
-          set.add((Authorizable) view);
+        val builder = AuthorizableImpl.builder();
```

Comment:
All of the built-in `Resource`s follow a `Viewable` -> `Resource.View` pattern which I didn't think a plugin developer should have to deal with. This code just maps a `Resource.AccessControlled` into an `Authorizable`, which is all the `Viewable` stuff was ever doing.

Reply:
Agree, nice call.